x-pack/packages/kbn-langchain/server/language_models/simple_chat_model.ts

```diff
@@ -150,18 +154,29 @@ export class ActionsClientSimpleChatModel extends SimpleChatModel {
     let finalOutputIndex = -1;
     const finalOutputStartToken = '"action":"FinalAnswer","action_input":"';
     let streamingFinished = false;
-    const finalOutputStopRegex = /(?<!\\)\"/;
+    const finalOutputStopRegex = /(?<!\\)"/;
+    let extraOutput = '';
```

Comment:
I noticed sometimes the first or last chunk was getting dropped. `extraOutput` captures any token that may be attached to the `finalOutputStartToken` or `finalOutputStopRegex `